lib/src/back_end/code_writer.dart

```diff
@@ -186,7 +190,12 @@ class CodeWriter {
     // to be used as the key in the memoization table.
     piece.format(this, state);
 
-    _pieceOptions.removeLast();
+    var childOptions = _pieceOptions.removeLast();
+
+    // If the child [piece] contains a newline then this one transitively does.
+    // TODO(tall): At some point, we may want to provide an API so that pieces
+    // can block this from propagating outward.
```

Comment:
Why would a piece want to block the propagation?

Reply:
I don't know if we'll need it, but the old IR has something similar to handle cases like:

```dart
function(argument, () {
  closure;
});
```
Usually, a newline inside an argument forces the argument list to split. But in this case, because it's a "block argument" we want to allow the newlines in the closure to *not* cause the surrounding argument list to split.

The new Piece architecture will, I think, have a better way of handling block arguments so we probably won't need to block propagation for this specific case, but there may be other places in the language where it's useful. The comment is basically a note to my future self in case it ends up being useful. Once more of the language is supported, if it turns out we didn't need it, I'll just remove the comment.